timesketch/api/v1/resources/scenarios.py

```diff
@@ -269,3 +270,96 @@ def post(self, sketch_id, scenario_id):
             db_session.commit()
 
         return self.to_json(scenario)
+
+
+class QuestionConclusionListResource(resources.ResourceMixin, Resource):
+    """Resource for investigative question conclusion."""
+
+    @login_required
+    def get(self, sketch_id, question_id):
+        """Handles GET request to the resource.
+
+        Returns:
+            A list of JSON representations of the conclusions.
+        """
+        sketch = Sketch.query.get_with_acl(sketch_id)
+        question = InvestigativeQuestion.query.get(question_id)
+
+        conclusions = InvestigativeQuestionConclusion.filter_by(
+            investigativequestion=question
+        ).all()
+
+        if not sketch:
+            abort(HTTP_STATUS_CODE_NOT_FOUND, "No sketch found with this ID")
+
+        return self.to_json(conclusions)
+
+    @login_required
+    def post(self, sketch_id, question_id):
+        """Handles POST request to the resource.
+
+        Adds or edits a conclusion.
+
+        Returns:
+            A JSON representation of the question.
+        """
+        sketch = Sketch.query.get_with_acl(sketch_id)
+        question = InvestigativeQuestion.query.get(question_id)
+
+        conclusion = InvestigativeQuestionConclusion.get_or_create(
+            user=current_user, investigativequestion=question
+        )
+
+        if not sketch:
+            abort(HTTP_STATUS_CODE_NOT_FOUND, "No sketch found with this ID")
+
+        if conclusion.user != current_user:
+            abort(HTTP_STATUS_CODE_FORBIDDEN, "User did not create this conclusion.")
+
+        form = request.json
+        if not form:
+            form = request.data
+
+        conclusion_text = form.get("conclusionText")
+        if conclusion_text:
+            conclusion.conclusion = conclusion_text
+            db_session.add(conclusion)
+            db_session.commit()
+
+        return self.to_json(conclusion)
+
+
+class QuestionConclusionResource(resources.ResourceMixin, Resource):
+    """Resource for investigative question conclusion."""
+
+    @login_required
+    def delete(self, sketch_id, question_id, conclusion_id):
+        """Handles DELETE request to the resource.
+
+        Deletes a conclusion.
+        """
+        sketch = Sketch.query.get_with_acl(sketch_id)
+        question = InvestigativeQuestion.query.get(question_id)
+        conclusion = InvestigativeQuestionConclusion.query.get(conclusion_id)
+
+        if not sketch:
+            abort(HTTP_STATUS_CODE_NOT_FOUND, "No sketch found with this ID")
+
+        if not question:
+            abort(HTTP_STATUS_CODE_NOT_FOUND, "No question found with this ID")
+
+        if not conclusion:
+            abort(HTTP_STATUS_CODE_NOT_FOUND, "No conclusion found with this ID")
```

Comment:
Same as above. We would save some queries if we check directly after each get and abort before continue to query the next element.

Reply:
fixed.